src/pages/Defi/hooks/useEarnBalances.tsx

```diff
@@ -22,32 +20,17 @@ export type UseEarnBalancesReturn = {
 
 export function useEarnBalances(): UseEarnBalancesReturn {
   const { isLoading: isFoxyBalancesLoading, data: foxyBalancesData } = useFoxyBalances()
-  const { cosmosSdkStakingOpportunities, totalBalance: totalCosmosStakingBalance } =
-    useCosmosSdkStakingBalances({
-      assetId: cosmosAssetId,
-    })
-  const {
-    cosmosSdkStakingOpportunities: osmosisStakingOpportunities,
-    totalBalance: totalOsmosisStakingBalance,
-  } = useCosmosSdkStakingBalances({
-    assetId: osmosisAssetId,
-  })
 
-  const stakingOpportunities = useAppSelector(selectAggregatedEarnUserStakingOpportunities)
+  const stakingOpportunities = useAppSelector(
+    selectAggregatedEarnUserStakingOpportunitiesIncludeUndelegations,
+  )
 
   const lpOpportunities = useAppSelector(selectAggregatedEarnUserLpOpportunities)
 
-  const stakingContractsAggregatedOpportunities = useAppSelector(
-    selectAggregatedEarnUserStakingOpportunities,
-  )
-
-  const farmContractsFiatBalance = useMemo(
+  const stakingOpportunitiesFiatBalance = useMemo(
```

Comment:
not so worried about this logic as this whole hook can die and go to selector heaven soon